internal/store/pod.go

```diff
@@ -278,6 +278,27 @@ var (
 				}
 			}),
 		},
+		{
+			Name: "kube_pod_failure_information",
+			Type: metric.Gauge,
+			Help: "The failed pods failure information (reason and message).",
+			GenerateFunc: wrapPodFunc(func(p *v1.Pod) *metric.Family {
+				ms := []*metric.Metric{}
+
+				phase := p.Status.Phase
+				if phase == v1.PodFailed {
+					ms = append(ms, &metric.Metric{
+						LabelKeys:   []string{"reason", "message"},
+						LabelValues: []string{p.Status.Reason, p.Status.Message},
```

Comment:
We can't use message in the metrics and if we are to use `Reason`, we need to maintain a known list of Pod failure reasons to prevent potential high cardinality problems.

A similar approach is used in metrics like
i) kube_pod_container_status_terminated
ii) kube_pod_container_status_waiting

Reply:
Hi @tariq1890 
Thanks for your review.

Afaik after digging into the code of the Kubernetes project, I could not find an exhaustive list of reasons for failed pods. Also, could you please explain why message are not allowed into metrics? I'm just curious about this. 

Anyway, it was nice to work on this great project. I will close this PR once you replied because I think it will not be possible to merge it according to the previous points.

All the best,
Alan

Reply:
The potential for high cardinality is higher when we record message strings. Messages are very much unbounded and it is not possible or practical to set a known list for that. Which Is why it makes sense to just have a known list of reasons rather than the message.

Reply:
By unbounded, I mean that the possible permutations of different messages are very high and can thus lead to high cardinality.

Reply:
While I don’t think we should close the PR right away, we could still having a metric like “kube_pod_status_failed”.

I am just waiting on the other maintainers to confirm this.

Reply:
@tariq1890 we have the `kube_pod_status_phase` with label `"phase"="Failed"`, just doesn't include the failure msg.

```
kube_pod_status_phase{namespace="default",pod="pod0",phase="Failed"} 0
```

Reply:
Whoops. Forgot about that :). Thanks